CRM/Contribute/Form/ContributionBase.php

```diff
@@ -936,23 +935,14 @@ protected function buildPremiumsBlock(bool $formItems = FALSE, $selectedOption =
     $products = [];
     $premium = [];
     foreach ($premiumProducts as $premiumProduct) {
-      $product = ['options' => NULL];
-      foreach ($premiumProduct as $key => $value) {
-        if (str_starts_with($key, 'product_id.')) {
-          if ($key === 'product_id.options' && $selectedProductID === $product['id'] && $selectedOption) {
-            // In this case we are on the thank you or confirm page so assign
-            // the selected option to the page for display.
-            $product['options'] = ts('Selected Option') . ': ' . $selectedOption;
-          }
-          else {
-            $product[str_replace('product_id.', '', $key)] = $value;
-          }
-        }
-        if (str_starts_with($key, 'premiums_id.')) {
-          $premium[str_replace('premiums_id.', '', $key)] = $value;
-        }
+      $product = CRM_Utils_Array::filterByPrefix($premiumProduct, 'product_id.');
+      $premium = CRM_Utils_Array::filterByPrefix($premiumProduct, 'premiums_id.');
```

Comment:
@eileenmcnaughton allow me to introduce a very handy function which takes the place of this entire loop!

Reply:
@eileenmcnaughton I cleaned this function up to get tests to pass (now that api4 returns an array for `PremiumProduct.options`) but I noticed one thing: when we extract `$product` from the api results it gets added to `$products[]` but when we extract `$premium` it overwrites `$premium`. Is that because there is only ever one premium regardless of the number of results?

Reply:
gawd I can never recall what the premium is vs the product...

I think you are right - I think only one premium can be attached to a given contribution page & that is basically 'THE' entity

Reply:
@eileenmcnaughton ok then this keeps the same behavior as before & is passing tests.